src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -6,6 +6,17 @@
 
 namespace System.Text.Json
 {
+    internal class JsonPreservedReference<T>
+    {
+        public JsonPreservedReference()
+        {
+            Values = default;
+            T copy = Values;
+        }
+
+        public T Values { get; set; }
```

Comment:
@stephentoub could you give advice on how to apply `PreserveDependencyAttribute` on a property getter and setter?

Reply:
What is the dependency you're trying to preserve? i.e. I think you're trying to ensure that get_Value and set_Value aren't removed if they may be needed... what would need them? What's accessing them via reflection? Etc.

Reply:
As you can see [here](https://github.com/dotnet/runtime/pull/655/files#diff-7d074ddc1c16d67315f2399ce9e1c2a6R108), when I step into a JSON Array that starts with curly brace `{` I need to switch to this `JsonPreserveReference<T>`, where `T` is the original array type containing `T Values` properties that is the returning enumerable.

Reply:
In that case you'd want to add an attribute like this to HandleStartObjectRef:
```C#
[PreserveDependency("get_Values", "System.Text.Json.JsonPreservedReference`1")]
[PreserveDependency("set_Values", "System.Text.Json.JsonPreservedReference`1")]
```

Reply:
I get it, thanks! I thought you had to add the attribute on top of the member you wanted to preserved.